main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"github.com/tidwall/gjson"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func validateCurrencyCode(CurrencyCode string) bool {
+	// description: validates currency code
+	result := false
+	// todo: work up a real test
+	if CurrencyCode == "is in list of currencies" {
+		result = true
+	} else {
+		result = false
+	}
+	return result
+}
+
+func getRate(CurrencyFrom string, CurrencyTo string) float64 {
+	httpposturl := fmt.Sprintf("https://query1.finance.yahoo.com/v7/finance/chart/%v%v%v", CurrencyFrom, CurrencyTo, "=x?corsDomain=finance.yahoo.com&range=1d&interval=1d")
```

Comment:
Lolled at `corsDomain=finance.yahoo.com`... the endpoint is not meant for API use I guess?

Reply:
:) 
the corsDomain variable was pulled from another API... not sure if it was on stackoverflow or github. Looks like it's not needed, will remove it:
`curl "https://query1.finance.yahoo.com/v7/finance/chart/USDCZK=x?range=1d&interval=1d"`